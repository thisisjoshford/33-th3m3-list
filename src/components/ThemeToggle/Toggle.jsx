import React from 'react';
import PropTypes from 'prop-types';

const Toggle = ({ toggle, value }) => (
  <>
    <input id="toggle" type="checkbox" checked={value} onChange={toggle} />
    <label>Dark</label>
  </>

);

Toggle.propTypes = {
  toggle: PropTypes.func.isRequired,
  value: PropTypes.bool.isRequired
};

export default Toggle;
