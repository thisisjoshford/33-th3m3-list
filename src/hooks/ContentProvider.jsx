import React, { createContext, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { getPeople } from '../services/apiFetch';

export const XfilesContent = createContext();

export const ContentProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);

  useEffect(() => {
    getPeople()
      .then(peeps => setCharacters(peeps));
  }, []);

  return (
    <XfilesContent.Provider value={characters}>
      {children}
    </XfilesContent.Provider>
  );
};

ContentProvider.propTypes = {
  children: PropTypes.node
};

