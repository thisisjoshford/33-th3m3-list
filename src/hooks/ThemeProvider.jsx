import React, { createContext, useState, useContext } from 'react';
import PropTypes from 'prop-types';

const ThemeContext = createContext();

export const ThemeProvider = ({ children }) => {
  const [theme, setTheme] = useState('light');
  const toggle = ({ target }) => {
    if(target.checked) setTheme('dark');
    if(!target.checked) setTheme('light');
  };

  return (
    <ThemeContext.Provider value={{ theme, toggle }}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.node
};

export const useTheme = () => {
  const theme = useContext(ThemeContext);
  return theme;
};

export const useToggle = () => {
  const toggle = useContext(ThemeContext);
  return toggle;
};
