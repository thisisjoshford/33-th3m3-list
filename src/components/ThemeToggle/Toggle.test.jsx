import React from 'react';
import { shallow } from 'enzyme';
import Toggle from './Toggle';
import { ThemeProvider } from '../../hooks/ThemeProvider';
import { ContentProvider } from '../../hooks/ContentProvider';

describe('Toggle component', () => {


  it('renders Toggle', () => {
    const wrapper = shallow(
      <ContentProvider>
        <Toggle value={true} toggle={()=>{}}/>
      </ContentProvider>
   
    );
    expect(wrapper).toMatchSnapshot();
  });
});
