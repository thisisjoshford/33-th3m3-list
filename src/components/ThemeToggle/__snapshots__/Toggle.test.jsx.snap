// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle component renders Toggle 1`] = `
<ContextProvider
  value={
    Object {
      "characters": Array [],
      "page": 1,
      "paging": [Function],
    }
  }
>
  <Toggle
    toggle={[Function]}
    value={true}
  />
</ContextProvider>
`;
